components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,589 @@
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _MAX_L1_MSG_FEE_`:: The maximum fee, in Wei, that Starknet #accepts/charges?# per single message.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L31[`getMaxL1MsgFee`]
+
+'''
+
+[#l1ToL2MessageCancellations]
+=== `l1ToL2MessageCancellations`
+
+[discrete]
+==== Description
+
+Returns the timestamp at the time cancelL1ToL2Message was called with a given message hash.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The message hash.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _result_`:: The timestamp at the time cancelL1ToL2Message was called with the given message hash `_msgHash_`. Returns 0 if cancelL1ToL2Message was never called with the given message hash `_msgHash_`.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L73[`l1ToL2MessageCancellations`]
+
+'''
+
+[#l1ToL2MessageNonce]
+=== `l1ToL2MessageNonce`
+
+[discrete]
+==== Description
+
+Returns the nonce of the next message sending to L2 contract.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+None.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _nonce_`:: The nonce of the next message sending to L2 contract.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L55[`l1ToL2MessageNonce`]
+
+'''
+
+[#l1ToL2Messages]
+=== `l1ToL2Messages`
+
+[discrete]
+==== Description
+
+Indicates if a pending message is associated with a given message hash.
+
+[discrete]
+==== State Mutability
+
+`view`
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The message hash.
+
+[discrete]
+==== Returns
+
+Returns `uint256 _result_`, where `_result_` is one of the following:
+[horizontal,labelwidth="30",role=stripes-odd]
+`msg_fee + 1`:: A pending message is associated with the `_msgHash_` parameter. #What is `msg_fee`? I don't see it mentioned anywhere. Does it simply refer to the fee of the pending message? Why does it return `msg_fee + 1`? How is that value used?#
```

Comment:
@LucasLvy Please respond.

Reply:
// Note that the inclusion of the unique nonce in the message hash implies that
// l1ToL2Messages()[msgHash] was not accessed before.
l1ToL2Messages()[msgHash] = msg.value + 1;

refered from https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/starknet/solidity/StarknetMessaging.sol#L121C1-L123C51